profiling/src/profiling/mod.rs

```diff
@@ -1034,43 +1040,44 @@ impl Profiler {
 
         let mut sample_types = Vec::with_capacity(SAMPLE_TYPES.len());
         let mut sample_values = Vec::with_capacity(SAMPLE_TYPES.len());
+
         if locals.profiling_enabled {
             // sample, wall-time, cpu-time
             let len = 2 + locals.profiling_experimental_cpu_time_enabled as usize;
             sample_types.extend_from_slice(&SAMPLE_TYPES[0..len]);
             sample_values.extend_from_slice(&values[0..len]);
+        }
 
-            // alloc-samples, alloc-size
-            if locals.profiling_allocation_enabled {
-                sample_types.extend_from_slice(&SAMPLE_TYPES[3..5]);
-                sample_values.extend_from_slice(&values[3..5]);
-            }
+        // alloc-samples, alloc-size
+        if locals.profiling_allocation_enabled {
```

Comment:
If profiling isn't enabled, then none of these following ifs need to be taken. I'm pretty sure that's why it was done the way it was before.